src/libsyncengine/update_detection/file_system_observer/remotefilesystemobserverworker.cpp

```diff
@@ -474,225 +478,241 @@ ExitCode RemoteFileSystemObserverWorker::sendLongPoll(bool &changes) {
-                        return exitCode;
-                    }
-                }
-            } else if (action == renameAction) {
-                _syncPal->removeItemFromTmpBlacklist(id, ReplicaSideRemote);
-                if (_snapshot->updateItem(item)) {
-                    LOGW_SYNCPAL_INFO(_logger, L"File/directory: " << SyncName2WStr(name).c_str() << L" ("
-                                                                   << Utility::s2ws(id).c_str() << L") renamed");
-                }
-            } else if (action == editAction) {
-                if (_snapshot->updateItem(item)) {
-                    LOGW_SYNCPAL_INFO(_logger, L"File/directory: " << SyncName2WStr(name).c_str() << L" ("
-                                                                   << Utility::s2ws(id).c_str() << L") edited at:" << modtime);
-                }
-            } else if (action == trashAction || action == moveOutAction || action == deleteAction || rightsRemoved) {
-                if (action == moveOutAction && movedItems.find(id) != movedItems.end()) {
-                    // Ignore move out action
-                    continue;
-                }
+        // Item edited
+        case ActionCode::actionCodeEdit:
+            _snapshot->updateItem(item);
+            break;
 
-                _syncPal->removeItemFromTmpBlacklist(id, ReplicaSideRemote);
-                if (_snapshot->removeItem(id)) {
-                    if (ParametersCache::instance()->parameters().extendedLog()) {
-                        LOGW_SYNCPAL_DEBUG(_logger, L"Item removed from remote snapshot: " << SyncName2WStr(name).c_str() << L" ("
-                                                                                           << Utility::s2ws(id).c_str() << L")");
-                    }
-                } else {
-                    LOGW_SYNCPAL_WARN(_logger, L"Fail to remove item: " << SyncName2WStr(name).c_str() << L" ("
-                                                                        << Utility::s2ws(id).c_str() << L")");
-                    invalidateSnapshot();
-                    return ExitCodeBackError;
-                }
-            } else if (action == accessAction || action == restoreFileShareCreate || action == restoreFileShareDelete ||
-                       action == restoreShareLinkCreate || action == restoreShareLinkDelete) {
-                // Ignore these actions
-            } else {
-                LOGW_SYNCPAL_DEBUG(_logger, L"Unknown operation received on file: " << SyncName2WStr(name).c_str() << L" ("
-                                                                                    << Utility::s2ws(id).c_str() << L")");
+        // Item removed
+        case ActionCode::actionCodeAccessRightRemove:
+        case ActionCode::actionCodeAccessRightUserRemove:
+        case ActionCode::actionCodeAccessRightTeamRemove:
+        case ActionCode::actionCodeAccessRightMainUsersRemove: {
+            bool hasRights = false;
```

Comment:
This case looks like the `ActionCode::actionCodeAccessRightMainUsersUpdate` case. Should they be handled the same way? If not, could we avoid code duplication in some way?

Reply:
The differences are on those lines:
```if (!hasRights) break;  // Current user does not have the right to access this item, ignore action.```
```if (hasRights) break;  // Current user still have the right to access this item, ignore action.```

And also those cases does not break but fall through the next statement, which is why they must be handled separately.
But I agree that there is some duplication here. Any suggestions are welcome ^^ I did not find a better/clearer way yet

Reply:
I would use a lambda function of the following form: 

```
const auto hasRights = [this](NodeId &nodeId) -> bool {
            SyncTime createdAt = 0;
            SyncTime modtime = 0;
            int64_t size = 0;
            if (getFileInfo(nodeId, hasRights, createdAt, modtime, size) != ExitCodeOk) {
               ...
}
```

and then call twice

`if (hasRights(actionInfo.nodeId)) ...`

with different side effects (break of fall through).

Reply:
Done 👍